From: Armbian <build@armbian.com>
Date: Tue, 16 Dec 2024 00:00:00 +0900
Subject: [PATCH] arm64: dts: sun50i-h616: H20 Pro 5G - Add PHY configuration for emac0

H20 Pro 5G uses emac0 with RMII PHY. Add proper PHY configuration to enable
network interface detection and initialization.

Signed-off-by: Armbian <build@armbian.com>
---
 arch/arm64/boot/dts/allwinner/sun50i-h616-h20-pro-5g.dts | 32 +++++++++++++++++++++++++++++---
 1 file changed, 29 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616-h20-pro-5g.dts b/arch/arm64/boot/dts/allwinner/sun50i-h616-h20-pro-5g.dts
index XXXXXXX..YYYYYYY 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616-h20-pro-5g.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-h20-pro-5g.dts
@@ -4,6 +4,7 @@
  */
 
 /dts-v1/;
 #include "sun50i-h616.dtsi"
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
@@ -11,6 +12,7 @@
 / {
 	model = "H20 Pro 5G";
 	compatible = "hechuang,h20-pro-5g", "allwinner,sun50i-h616";
+	ethernet0 = &emac0;
 
 	aliases {
 		serial0 = &uart0;
@@ -26,3 +28,35 @@
 		rootfstype=ext4 earlycon=uart,mmio32,0x02500000
 	};
 };
+
+&emac0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&rmii_phy>;
+	phy-supply = <&reg_dldo1>;
+	allwinner,rx-delay-ps = <3100>;
+	allwinner,tx-delay-ps = <700>;
+	status = "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	rmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+		max-speed = <100>;
+		clocks = <&ccu CLK_AC200_PHY>;
+		clock-names = "refclk";
+		reset-gpios = <&pio 3 14 GPIO_ACTIVE_LOW>; /* PD14 */
+		reset-assert-us = <75000>;
+		reset-deassert-us = <15000>;
+	};
+};
+
+&emac1 {
+	status = "disabled";
+};
+
